keyboards/black_hellebore/rules.mk

```diff
@@ -0,0 +1,10 @@
+############################################################################
+#
+# PLL of 72Mhz / 32 = 2.25Mhz
+# PLL of 72Mhz / 2 = 36Mhz (This shouldn't work, but does)
+#
+
+RGB_MATRIX_DRIVER = WS2812
+WS2812_DRIVER = spi
+WS2812_SPI_DIVISOR = 2		
+
```

Comment:
Not sure where you decided that the PLL is 72MHz, but it's incorrect.
The L412 board files are set up with an 80MHz system clock, which propagates to all other peripherals as well, i.e. no subsequent clock divisors.

Specifying `WS2812_SPI_DIVISOR = 2` here has no effect whatsoever -- it'd need to be in code, not makefile. As it stands, you're getting the default -- `/16`, so `5 MHz` SPI. Suggest removal, if it's working -- otherwise you might have better luck swapping to PWM output. A7 has `TIM1_CH1N` output -- see `#define WS2812_PWM_COMPLEMENTARY_OUTPUT` in the docs.

Reply:
Then what is `#define STM32_PLLSAI1N_VALUE                72` ? (I'm not trying to be 'smart', I honestly thought that was the config for the internal oscillator)

Noted, I'll move it to the make file, I appreciate the time you took to respond. I'll check out `WS2812_PWM_COMPLEMENTARY_OUTPUT`, but can you use PWM on chip led's? the led data sheet specs it as a digital pulse train.

Reply:
OH! I completely over looked that part of the doc, that's way better! Thanks for the advice.

Reply:
SAI is the serial audio interface, and is not enabled within QMK.
[These](https://github.com/qmk/qmk_firmware/blob/master/platforms/chibios/boards/GENERIC_STM32_L412XB/configs/mcuconf.h#L52-L62) are the only values in play for your board.

Reply:
> can you use PWM on chip led's? the led data sheet specs it as a digital pulse train.

Yes, there's a driver specifically set up for the required timing.